src/main/java/executor/service/service/impl/ProxySourcesClientImpl.java

```diff
@@ -1,11 +1,48 @@
 package executor.service.service.impl;
 
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import executor.service.config.properties.PropertiesConstants;
 import executor.service.model.ProxyConfigHolder;
 import executor.service.service.ProxySourcesClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.List;
+
 
 public class ProxySourcesClientImpl implements ProxySourcesClient {
```

Comment:
1. Reading ProxyNetworkConfig, ProxyCredentials from different files: proxy-network.json, proxy-credential.json:
![image](https://github.com/GFLCourses5/worker/assets/140727960/3b194a14-f16c-477f-957f-fb129403b613)

2. As a result, it is necessary to ensure that the read ProxyNetworkConfig and ProxyCredentials are correctly combined into the ProxyConfigHolder (the problem may be a different number of elements in the proxy-network.json, proxy-credential.json files).

3. It is also necessary to simulate the situation of random data arrival for implementation in ParralelFlowExecutorService.

In my opinion, pull-request#43 is more suitable for these tasks.

Reply:
1. That's one way of solving this task. In my opinion, not the best one

2. That's the main reason why I propose to combine these files into one. Either one has no use without other - ProxyNetworkConfig will not be valid without proper credentials; credentials will have no use without ProxyNetworkConfig

3. I suppose, ParralelFlowExecutorService consumes ProxyConfigHolder, so I don't get your point. Also, proxies come through validation before being added to list

Reply:
I just noticed the annul contract.